src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -954,18 +1045,21 @@ void VTKMeshGenerator::generateMesh( DomainPartition & domain )
   // TODO refactor void MeshGeneratorBase::generateMesh( DomainPartition & domain )
   GEOSX_MARK_FUNCTION;
 
-  vtkSmartPointer< vtkMultiProcessController > controller = getVTKController();
+  MPI_Comm const comm = MPI_COMM_GEOSX;
+  vtkSmartPointer< vtkMultiProcessController > controller = vtk::getController();
   vtkMultiProcessController::SetGlobalController( controller );
 
   GEOSX_LOG_RANK_0( GEOSX_FMT( "{} '{}': reading mesh from {}", catalogName(), getName(), m_filePath ) );
   {
-    vtkSmartPointer< vtkUnstructuredGrid > loadedMesh = loadVTKMesh( m_filePath );
-    std::vector< vtkBoundingBox > cuts;
-    m_vtkMesh = redistributeMesh( *loadedMesh, cuts );
-
-    // TODO Check that the neighbor information set is bulletproof
-    std::unordered_set< int > const neighbors = computeMPINeighborRanks( std::move( cuts ) );
+    GEOSX_LOG_LEVEL_RANK_0( 2, "  reading the dataset..." );
+    vtkSmartPointer< vtkUnstructuredGrid > loadedMesh = vtk::loadMesh( m_filePath );
+    GEOSX_LOG_LEVEL_RANK_0( 2, "  redistributing mesh..." );
+    m_vtkMesh = vtk::redistributeMesh( *loadedMesh, comm, m_partitionRefinement );
+    GEOSX_LOG_LEVEL_RANK_0( 2, "  finding neighbor ranks..." );
+    std::vector< vtkBoundingBox > boxes = vtk::exchangeBoundingBoxes( *m_vtkMesh, comm );
+    std::vector< int > const neighbors = vtk::findNeighborRanks( std::move( boxes ) );
     domain.getMetisNeighborList().insert( neighbors.begin(), neighbors.end() );
+    GEOSX_LOG_LEVEL_RANK_0( 2, "  done!" );
```

Comment:
Thanks for all the logging that has been added to the mesh generators over multiple PRs, it is saving a lot of debugging time